accessors/clients/cassandra/interface.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package cassandraclient
+
+import (
+	"fmt"
+	"github.com/gocql/gocql"
+)
+
+type GocqlSessionInterface interface {
+	KeyspaceMetadata(keyspace string) (*gocql.KeyspaceMetadata, error)
+	Close()
+}
+
+type KeyspaceMetadataInterface interface {
+	Tables() map[string]*gocql.TableMetadata
+}
+
+type CassandraClusterInterface interface {
+	KeyspaceMetadata(keyspace string) (KeyspaceMetadataInterface, error)
+	Close() 
+}
+
+type GocqlSessionImpl struct {
+	session *gocql.Session
+}
```

Comment:
Can we add UT to the functions here?

I guess it should be possible to construct this with mock session and verify all the branches of if-else.